### ISSUE TITLE
Add interfaces for MVI & extended concepts

### ISSUE THREAD

**Issue by chikamichi** [chikamichi/typometer#43]
- `Actions` => `{[key: string]: Stream<any>}`
- `Model` => `(actions: Actions, state$?: MemoryStream<State>): Stream<Reducer>`
- `View` => `(sources: Sources): Stream<VNode>`
- `Intent` => `(domSource: DOMSource): Actions`
- `NAP` => `(state$: MemoryStream<State>): Actions`

**Comment by chikamichi** [comment]
Refactor code, such as for instance a view is defined with:

``` ts
const View: View = (state$) => {
  return state$.map(…);
}

export default View
```

**Comment by chikamichi** [comment]
Implemented in 6219a7d2878834a81c68f585b284ac909f136eb7.
